lisdf/parsing/pddl_j.py

```diff
@@ -144,30 +148,34 @@ def typedvariable(self, name, typename):
     def constant(self, name):
         name = name.value
         sdf_object = None
-        if name in self.sdf.model_dict:
-            sdf_object = C.PDDLSDFObject(name, None, self.domain.types["sdf::model"])
-        elif name in self.sdf.link_dict:
-            if NAME_SCOPE_SEP is None:
-                model_name, lname = "", name
-            else:
-                model_name, lname = name.split(NAME_SCOPE_SEP)
-            sdf_object = C.PDDLSDFObject(
-                model_name, lname, self.domain.types["sdf::link"]
-            )
-        elif name in self.sdf.joint_dict:
-            if NAME_SCOPE_SEP is None:
-                model_name, lname = "", name
-            else:
-                model_name, jname = name.split(NAME_SCOPE_SEP)
-            sdf_object = C.PDDLSDFObject(
-                model_name, jname, self.domain.types["sdf::joint"]
-            )
+        if self.sdf is not None:
+            if name in self.sdf.model_dict:
+                sdf_object = C.PDDLSDFObject(
+                    name, None, self.domain.types["sdf::model"]
+                )
+            elif name in self.sdf.link_dict:
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
+                else:
+                    model_name, lname = name.split(NAME_SCOPE_SEP)
+                sdf_object = C.PDDLSDFObject(
+                    model_name, lname, self.domain.types["sdf::link"]
+                )
+            elif name in self.sdf.joint_dict:
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
+                else:
+                    model_name, jname = name.split(NAME_SCOPE_SEP)
+                sdf_object = C.PDDLSDFObject(
+                    model_name, jname, self.domain.types["sdf::joint"]
+                )
 
         type = None
         if sdf_object is not None:
             type = sdf_object.sdf_type
         if name in self.problem.objects:
             type = self.problem.objects[name].type
+
         return C.PDDLObject(name, type, sdf_object=sdf_object)
```

Comment:
If `type is None and sdf_object is None` do we want to return an essentially empty object here or raise value error?